File: src/main/java/com/45secondapps/service/BillingService.java
package com.45secondapps.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.45secondapps.domain.Billing;
import com.45secondapps.dto.BillingDTO;
import com.45secondapps.dto.BillingSearchDTO;
import com.45secondapps.dto.BillingPageDTO;
import com.45secondapps.dto.BillingConvertCriteriaDTO;
import com.45secondapps.service.GenericService;
import com.45secondapps.dto.common.RequestDTO;
import com.45secondapps.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface BillingService extends GenericService<Billing, Integer> {

	List<Billing> findAll();

	ResultDTO addBilling(BillingDTO billingDTO, RequestDTO requestDTO);

	ResultDTO updateBilling(BillingDTO billingDTO, RequestDTO requestDTO);

    Page<Billing> getAllBillings(Pageable pageable);

    Page<Billing> getAllBillings(Specification<Billing> spec, Pageable pageable);

	ResponseEntity<BillingPageDTO> getBillings(BillingSearchDTO billingSearchDTO);
	
	List<BillingDTO> convertBillingsToBillingDTOs(List<Billing> billings, BillingConvertCriteriaDTO convertCriteria);

	BillingDTO getBillingDTOById(Integer billingId);







}

File: 45secondapps_ui/src/services/BillingService.js
import http from "../http-common"; 

class BillingService {
  getAllBillings(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/billing/billings`, searchDTO);
  }

  get(billingId) {
    return this.getRequest(`/billing/${billingId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/billing?field=${matchData}`, null);
  }

  addBilling(data) {
    return http.post("/billing/addBilling", data);
  }

  update(data) {
  	return http.post("/billing/updateBilling", data);
  }
  
  uploadImage(data,billingId) {
  	return http.postForm("/billing/uploadImage/"+billingId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new BillingService();
